### PR TITLE
[python3] Bump to 3.10.7

### PR DESCRIPTION
**Pull request by Hoikas**
Bump the python3 port to the latest revision of 3.10.x as a precursor to upgrading to 3.11.0. This front-loads some (but not all) of the tweaks that will need to be made to the patches for 3.11.0.

- #### What does your PR fix?
  - Updates python to 3.10.7.
  - Some patch files were diffs masquerading as patch files. Change those to actual patch files so they can be more easily applied onto the cpython repo.
  - Fixed a compile breakage in the Windows 7 patch introduced somewhere between 3.10.5 and 3.10.7.
  - Tweaked some of the static building patches to prevent conditional patch content appearing in unconditional context lines. This will be more relevant in 3.11.0, but I'm front loading it now.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  No change

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Indeed

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/python3/portfile.cmake [hunk 1]
```diff
@@ -5,24 +5,28 @@ endif()
...
 endif()
+
+list(APPEND PATCHES
+    0003-use-vcpkg-zlib.patch
+    0004-devendor-external-dependencies.patch
+    0005-dont-copy-vcruntime.patch
+    0008-python.pc.patch
+    0009-bz2d.patch
+    0010-dont-skip-rpath.patch
+)
```
Comment (dg0yt) [inline ports/python3/portfile.cmake:26]: I would prefer to have the unconditional patches before all the optional patches. And make patches unconditional where possible.

Comment (Hoikas) [reply ports/python3/portfile.cmake:26]: I prefer for patches to be applied in the same order that they are in my git history where possible. This eases my ability to work in this port in my spare time. I realize that not all of the patches are in this order, and I should probably fix that in this pull request.

Comment (Hoikas) [reply ports/python3/portfile.cmake:26]: Should be fixed now.

**Comment by Hoikas** [comment]
Everything is green, so we're ready for review.

File: ports/python3/0002-static-library.patch [hunk 2]
```diff
@@ -62,7 +75,7 @@ diff --git a/Python/sysmodule.c b/Python/sysmodule.c
 index ac49f7867a..f3583345ff 100644
 --- a/Python/sysmodule.c
 +++ b/Python/sysmodule.c
-@@ -2804,6 +2804,9 @@ _PySys_InitCore(PyThreadState *tstate, PyObject *sysdict)
+@@ -2804,6 +2804,9 @@ _PySys_InitCore(PyThreadState *tstate, PyObject *sysdict
```
Comment (JonLiu1993) [inline ports/python3/0002-static-library.patch:78]: @Hoikas, Thanks for your pr, I haven't looked into that, but it looks like there's a missing ")" here?

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@Hoikas, I have reviewed this pr, it is very good, I saw that this pr has a big change to the upstream code, I think it is necessary to let the upstream know. do you think so?

File: ports/python3/0002-static-library.patch (hunk 2 above)
Comment (dg0yt) [reply ports/python3/0002-static-library.patch:78]: This line is generated by `diff` and describes the context for the human readers. Or confuses the reader ;-)

**Comment by JonLiu1993** [comment]
Feature `deprecated-win7-support` tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

File: ports/python3/0002-static-library.patch (hunk 2 above)
Comment (Hoikas) [reply ports/python3/0002-static-library.patch:78]: Heh, yeah, reading diffs of diffs is difficult. Like @dg0yt mentioned, this is a context line that needed to be updated due to changes by upstream. the `)` was probably moved to a newline upstream.

**Comment by Hoikas** [comment]
> @Hoikas, I have reviewed this pr, it is very good, I saw that this pr has a big change to the upstream code, I think it is necessary to let the upstream know. do you think so?

All of these patches are specific for building Python in the vcpkg environment or in ways that are not particularly supported upstream (eg deprecated-win7-support), so I don't think so.

**Review by JonLiu1993 — APPROVED** [review]

**Review by BillyONeal — APPROVED** [review]
For what it's worth, we do not care about Windows 7: not even the server versions of that are in support anymore. Someone is willing to keep the patch here and I see no reason to go out of our way to break it but we won't hesitate to drop that if an update breaks it for some reason.

Thanks for the update!

**Pull request merged**